src/vellum/workflows/state/encoder.py

```diff
@@ -18,6 +20,22 @@
 from vellum.workflows.utils.functions import compile_function_definition
 
 
+def virtual_open(file_path: str, mode: str = "r"):
+    """
+    Open a file, checking VirtualFileFinder instances first before falling back to regular open().
+    """
+    for finder in sys.meta_path:
+        if hasattr(finder, "loader") and hasattr(finder.loader, "_get_code"):
```

Comment:
Works for now. Ideally we create a public interface that `VirtualFileFinder` extends from that we could use here and make the equivalent method to `_get_code` public